x-pack/solutions/observability/plugins/streams/server/routes/streams/processing/suggestions_handler.ts

```diff
@@ -0,0 +1,235 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { IScopedClusterClient } from '@kbn/core/server';
+import { get, uniqBy } from 'lodash';
+import { InferenceClient } from '@kbn/inference-plugin/server';
+import { RecursiveRecord } from '@kbn/streams-schema';
+import { ProcessingSuggestionParams } from './route';
+import {
+  assertSimulationResult,
+  executeSimulation,
+  prepareSimulationBody,
+  prepareSimulationDiffs,
+  prepareSimulationResponse,
+} from './simulation_handler';
```

Comment:
@flash1293 all of this will be gone and won't adapt to the updated simulation API. Relying on them would probably break this API, I suggest to copy them over and later on adapt this API to use the new simulation logic where necessary.

Reply:
Sounds good, let's do this - I can either adjust here or you can on your side, depending on what merges first

Reply:
Has this been applied? These utils should be copied here and not imported from the simulation_handler (since they won't exist anymore with https://github.com/elastic/kibana/pull/209985)